docs/rules/expiring-todo-comments.md

```diff
@@ -0,0 +1,262 @@
+- `[engine:node@>8]` or `[engine:node@>=8]` to expire when package bump supported engines. No whitespace.
+
+## Overall Information
+
+- Multiple conditions can be separated by comma:
+  `[YYYY-MM-DD, +foo-package, engine:node@>=8, bar-package@>=2]`.
+- You cannot stack multiple dates or multiple versions of
+  your package on the same TODO.
+- You can use either `TODO`, `FIXME`, or `XXX`. You can change
+  these using options. (See [options.terms](#terms)).
+- TODOs may have any string before arguments. Really useful
+  when you want to highlight more information such as author name
+  like `TODO (lubien) [2019-05-30]` or `TODO @lubien [2019-05-30]`.
+- TODOs may or may not have a colon before the message such as
+  `TODO [...]: message` or `TODO [...] message`.
+- If no proper argument is found, you'll be notified that the TODO is useless (See [`eslint/no-warning-comments`](#disallow-warning-comments-no-warning-comments)).
+
+## Fail
+
+```js
+// TODO [2000-01-01]: I'll fix this next week.
+// TODO [2000-01-01, 2001-01-01]: Multiple dates won't work.
+
+// TODO [>1]: If your package.json version is > 1
+// TODO [>=1]: If your package.json version is >= 1
+// TODO [>1, >2]: Multiple package versions won't work
+
+// TODO [+already-have-pkg]: Since we already have it, this reports.
+// TODO [-we-dont-have-this-package]: Since we don't have, trigger report.
+
+// TODO [read-pkg@>1]: When `read-pkg` version is > 1 don't forget to do this
+// TODO [read-pkg@>=5.1.1]: When `read-pkg` version is >= 5.1.1 don't forget to do that
+
+// TODO [engine:node@>=8]: Whoops we are already supporting it!
+
+// TODO: Add unicorns.
+```
+
+## Pass
+
+```js
+// TODO [2200-12-25]: Too long... Can you feel it?'
+// FIXME [2200-12-25]: Too long... Can you feel it?
+
+// TODO (lubien) [2200-12-12]: You can add something before the arguments.
+// TODO @lubien [2200-12-12]: You can add something before the arguments.
+// FIXME [2200-12-25] (lubien): You can add something after the arguments, before the colon.
+// TODO [2200-12-12] No colon after argument.
+
+// TODO [+react]: Refactor this when we use React.
+// TODO [-lodash]: If we remove lodash we need to change this.
+
+// TODO [lodash@>10]: Lodash has a new way to do this, when we bump version let's use it.
+// TODO [lodash@>=10]: Lodash has a new way to do this, when we bump version let's use it.
+
+// TODO [2200-12-25, +popura, lodash@>10]: Combo.
+
+// TODO [engine:node@>12]: When we bump version we can use import/export.
+
+/*
+ * TODO [2200-12-25]: Yet
+ * TODO [2200-12-25]: Another
+ * TODO [2200-12-25]: Way
+ */
+```
+
+## Options
+
+### ignoreDatesOnPullRequests
+
+Disables time checks during pull requests.
+
+Sometimes developers may send [Pull Requests](https://help.github.com/en/articles/about-pull-requests) right on time when TODO expiry dates are triggered. This will mean that their code would fail to pass linting and would cause a false positive.
```

Comment:
```suggestion
Sometimes developers may send [Pull Requests](https://help.github.com/en/articles/about-pull-requests) at a time when TODO expiry dates are triggered. This means that their code would fail to pass linting, causing a false positive.
```